routes/devices.js

```diff
@@ -45,17 +39,96 @@ devices.get('/json', auth.ensureAuthenticated, function (req, res) {
 devices.get('/:id', auth.ensureAuthenticated, function (req, res) {
   var deviceId = req.params['id']
   for( var idx=0; idx<devDocs.length; idx+=1 ) {
-    if( deviceId === devDocs[idx]._id ) {
-      res.render('device', { device: devDocs[idx] });
+    if( devDocs[idx]._id.equals(deviceId) ) {
+      res.render('device', { 
+        device: devDocs[idx],
+        checklistValues: checklistValues,
+        checklistSubjects: checklistSubjects
+      });
       return
     }
   }
   res.status(404).render('error', {
     error:{
       status:'Device not Found'
     }
-  })
-})
+  });
+});
+
+
+devices.post('/:id', auth.ensureAuthenticated, function (req, res) {
```

Comment:
we might need put the config of a checklist and update the checklist in different routes. 
PUT /:id/config/checklist
to initialize the checklist when the AM requests a checklist. This can be used for update the whole checklist if the client requests in such a way. 

PUT /:id/config/checklist/:subject
to update an individual subject (required or not)